pkg/provisioner/templates/kernel.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package templates
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/NVIDIA/holodeck/api/holodeck/v1alpha1"
+)
+
+const kernelTemplate = `
```

Comment:
This seems specific to Ubuntu. Should we have an interface for Kernel installation and distro-specifc implementations of the same?

Reply:
Yeah currently ALL templates in Holodeck are Ubuntu focused, is on my TODO to add a RPM path. not in scope for this PR tho

Reply:
Sure